chia/rpc/data_layer_rpc_api.py

```diff
@@ -458,3 +468,42 @@ async def clear_pending_roots(self, request: ClearPendingRootsRequest) -> ClearP
         root = await self.service.data_store.clear_pending_roots(tree_id=request.store_id)
 
         return ClearPendingRootsResponse(success=root is not None, root=root)
+
+    @streamable_marshal
+    async def get_proof(self, request: GetProofRequest) -> GetProofResponse:
+        root = await self.service.get_root(store_id=request.store_id)
+        if root is None:
+            raise ValueError("no root")
+
+        all_proofs: List[HashOnlyProof] = []
+        for key in request.keys:
+            key_value = await self.service.get_value(store_id=request.store_id, key=key)
+            pi = await self.service.data_store.get_proof_of_inclusion_by_key(tree_id=request.store_id, key=key)
+
+            proof = HashOnlyProof(
+                key_clvm_hash=Program.to(key).get_tree_hash(),
+                value_clvm_hash=Program.to(key_value).get_tree_hash(),
+                node_hash=pi.node_hash,
+                layers=[
+                    ProofLayer(
+                        other_hash_side=uint8(layer.other_hash_side),
+                        other_hash=layer.other_hash,
+                        combined_hash=layer.combined_hash,
+                    )
+                    for layer in pi.layers
+                ],
+            )
```

Comment:
It feels like a helper classmethod on `HashOnlyProof` might be worthwhile.  It both avoids repetition of the conversion code and defines the expected hash pattern at the class.